solr/solrj/src/java/org/apache/solr/common/util/ObjectCache.java

```diff
@@ -78,12 +84,26 @@ public <T> T computeIfAbsent(
 
   @Override
   public boolean isClosed() {
-    return isClosed;
+    return isClosed.get();
   }
 
   @Override
   public void close() throws IOException {
-    isClosed = true;
-    map.clear();
+    if (isClosed.compareAndSet(false, true)) {
```

Comment:
It's not clear to me changes in this file? is it separate issue?

Reply:
This was cherry-picked from upstream, and was initially the only change on this branch (before the TeeDirectory stuff moved in here). We depend on this to close node-level resources that are tracked in the ObjectCache (metrics, an ExecutorService, and if applicable the background eager loading task).

Reply:
https://github.com/cowpaths/fullstory-solr/pull/182/commits/654f882b601222238f411fa8e941fac62728e32e